backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -36,11 +36,25 @@ public class UserService : BaseService<User>, IUserService
         /// <param name="quantity"></param>
         /// <param name="sort"></param>
         /// <returns></returns>
-        public Paged<User> GetNoTracking(int page = 1, int quantity = 10, string sort = null)
+        public Paged<User> GetNoTracking(int page = 1, int quantity = 10, string sort = null, Guid? userId = null)
         {
+
             this.User.ThrowIfNotAuthorized("system-administrator");
+            var query = this.Context.Users
+                .Include(u => u.Agencies)
+                .ThenInclude(a => a.Agency)
+                .Include(r => r.Roles)
+                .ThenInclude(r => r.Role)
+                .AsNoTracking();
+            if (userId.HasValue) {
```

Comment:
Making this type of request constantly against the DB isn't ideal.  Just use the `User.GetAgencies()` extension method.  We'll include this information in their Claims.